### ISSUE TITLE
need feature selection?

### ISSUE THREAD

**Issue by dyd912** [microsoft/LightGBM#3650]
I am new to lightGBM and have been using lightGBM for prediction for several weeks. I checked feature importance after training the model. However, I am not sure whether deleting some "less importatnt" features can improve the test result or not. 
I have around 150 features and tons of samples, training an optimal model is time-consuming for me. Therefore, I want to know if feature selection can improve the test result.
Any idea is aprreciated.

**Comment by jameslamb** [comment]
hi @dyd912 , thanks for using LightGBM!

### Short Answer

> I want to know if feature selection can improve the test result.

LightGBM is a tree-based, supervised algorithm so it already has feature selection built into it. Because of this, I wouldn't expect dropping features to improve performance on a test set.

### Longer Answer

You should not expect the test performance to suffer from the presence of features that have low importance. But having more features does increase the training time.

The right balance between training speed and the number of features depends on your specific dataset and the problem you're trying to solve with a model.

See https://lightgbm.readthedocs.io/en/latest/Parameters-Tuning.html#for-faster-speed for a guide on how to change LightGBM's parameters to improve training time. 

The tradeoff here also depends on how complex your model is. Be sure that you have high enough `num_iterations`, `num_leaves`, and `max_depth` for the full feature space to be explored.
